lib/Wikia/src/Service/User/Attributes/AttributeKeyValueService.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace Wikia\Service\User\Attributes;
+
+use Wikia\Logger\Loggable;
+use Wikia\Persistence\User\Attributes\AttributePersistence;
+use Wikia\Util\WikiaProfiler;
+use Wikia\Service\User\Attributes;
+
+class AttributeKeyValueService implements AttributeService {
+
+	const PROFILE_EVENT = \Transaction::EVENT_USER_ATTRIBUTES;
+
+	use WikiaProfiler;
+	use Loggable;
+
+	/** @var AttributePersistence */
+	private $persistenceAdapter;
+
+	/**
+	 * @Inject({
+	 *    Wikia\Persistence\User\Attributes\AttributePersistence::class
+	 * })
+	 * @param AttributePersistence $persistenceAdapter
+	 */
+	function __construct( AttributePersistence $persistenceAdapter ) {
+		$this->persistenceAdapter = $persistenceAdapter;
+	}
+
+	public function setAttribute( $userId, $attribute ) {
+		if ( empty( $attribute ) || $userId == 0 ) {
```

Comment:
Why it's not possible to set empty value for attribute? 
I can imagine setting attribute flags - (```true/false``` value). In this case I cannot set false value. Additionally there is no way to delete attribute right now.